### PR TITLE
Invalid Index Detection

### PR DESCRIPTION
**Pull request by ozthegnp**
updating algorithm of .srt entry detection
fixing separators for repairable timecode syntax
Detecting invalid .srt indices
Detecting invalid entry at the beginning of .srt file.
Add test cases:
    - 'should catch non-numerical indices'
    - 'should catch entry without timecode'

### CODE REVIEW

File: lib/parsers/srtEntriesRegex.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const validMinutes = '[0-5]\\d';
+const validSeconds = '[0-5]\\d';
+const validMilliseconds = '\\d{3}';
+const any = '[^\\n\\d]+?'; // Any character but digit or newline
+const validTimestamp = `${validHours}${any}${validMinutes}${any}${validSeconds}${any}${validMilliseconds}`;
+const validTimecode = `${validTimestamp}[^\\n\\d]+?${validTimestamp}[^\\n\\d]*?`;
+const textOfEntry = '[^]+';
+
+// Capturing Groups
+const Group1 = `(${validTimestamp})`;
```
Comment (McCourtRC) [inline lib/parsers/srtEntriesRegex.js:20]: is this supposed to be `validIndex` ?

Comment (McCourtRC) [inline lib/parsers/srtEntriesRegex.js:20]: is this supposed to be `validIndex`?

File: lib/parsers/srtEntries.js [hunk 2]
```diff
@@ -1,93 +1,105 @@
...
+  }
+
+  // Validating all entries and creating return objects
+  const { validEntries, status } = potentialBlocksArray.reduce((acc, cur) => {
+    // Remove excessive newlines
+    cur = cur.replace(/\n{2,}/g, '\n').trim();
+
+    // Check if index exist and valid
+    const potentialIndex = cur.match(potentialIndexRegex);
+    if (potentialIndex && !(/\d+/.test(potentialIndex))) {
```
Comment (McCourtRC) [inline lib/parsers/srtEntries.js:88]: Do we need to be stricter when identifying a valid index?  Should the line be only digits? `/^\d+/$`

File: lib/parsers/srtEntries.js [hunk 3]
```diff
@@ -1,93 +1,105 @@
...
+
+/**
+ * Functions for parseSrtEntries
+ */
+function standardizeTimestamp(timestamp) {
+  if (strictTimestampRegex.test(timestamp)) return timestamp;
+  return timestamp.replace(/[^\d]+/g, ':').replace(/:(?=\d{3})/, ','); // HH:MM:SS,mmm format
+}
+
+function pushInvalidEntry(acc, cur, options) {
```
Comment (McCourtRC) [inline lib/parsers/srtEntries.js:21]: let's try to keep the invalid entry as an object so it is easy to identify the different pieces: `{idx, timecode, text}`.  Any missing values can be `undefined`

**Comment by ozthegnp** [comment]
@McCourtRC all suggestions have been added to the latest commit

**Pull request merged**